components/higher-order/with-api-data/index.js

```diff
@@ -159,16 +153,14 @@ export default ( mapPropsToData ) => createHigherOrderComponent( ( WrappedCompon
 				} );
 		}
 
-		applyMapping( props ) {
-			const { dataProps } = this.state;
-
+		applyMapping( props, previousDataProps = {} ) {
```

Comment:
Would it be much trouble to factor this into a `static getDerivedStateFromProps` ? Main concern is the double render by `componentDidUpdate` calling `setState`.

Reply:
I tried this first but it's not possible at the moment because we rely on `context` and there's no way we can access `context` in `getDerivedStateFromProps`